crates/notion-core/src/session.rs

```diff
@@ -203,9 +203,19 @@ impl Session {
     /// Fetch, unpack, and install a package matching the input requirements.
     pub fn install_package(&mut self, name: String, version: &VersionSpec) -> Fallible<Version> {
         // fetches and unpacks package
-        let fetched_package = self.fetch_package(name, version)?;
+        let fetched_package = self.fetch_package(name.clone(), version)?;
```

Comment:
I'm not sure of the exact way to resolve it (if there even is one), but it seems like there are a lot of calls to `clone` in this method. In my (admittedly limited) experience with Rust, that tends to point to the data model not being ideal. It's not a big issue or a blocker for this PR, especially since the objects we're cloning here are small (versions and Strings), but just something that stuck out to me.

Perhaps @dherman or @chriskrycho can weigh in, having significantly more Rust knowledge?

Reply:
Two notes:

1. We can almost certainly plumb references down through as need be to make it work to just use references here (though I'd file that as a tech debt issue)—it should be fairly mechanical, and in fact I don't see (on an initial perusal) anywhere we *can't* just use an `&str` instead that we're passing down `String`s through this chain currently.

2. It's true that it's nice (and generally preferable) to avoid `clone`-ing unnecessarily, but that's often best left for cleanup/optimization phases. We'll use *slightly* more memory by allocating copies of this `String`, but it's going to be a pretty small amount and is very unlikely to be especially meaningful in the use of the tool. It's a good thing to do, but it's context-dependent how worthwhile it is to invest there.

Reply:
I just checked… and the Rust compiler is so ridiculously good; it was indeed trivial to plumb this through. [Opened a PR against this branch](https://github.com/mikrostew/notion/pull/1) that you can merge or ignore as you like, @mikrostew!

Reply:
I, uhh, will neither confirm nor deny reports that I got carried away and removed *a bunch* of moves and allocations on that branch/PR. 😛

Reply:
Thanks @chriskrycho, just merged your PR. That cleans this up nicely